### PR TITLE
Fixing UA matching for Safari

### PR DESCRIPTION
**Pull request by aghassemi**
Closes https://github.com/ampproject/amphtml/issues/4374

### CODE REVIEW

File: test/_init_tests.js [hunk 1]
```diff
@@ -106,6 +106,18 @@ class TestConfig {
       return fn.apply(this, arguments);
     });
   }
+
+  /** @private */
+  isAgentMatched_(agent) {
```
Comment (jridgewell) [inline test/_init_tests.js]: Can we reuse `platform.js`?

Comment (aghassemi) [inline test/_init_tests.js]: oh didn't know about `platform.js`. Yeah, should be able too. Wait for the next patch before reviewing..

**Comment by aghassemi** [comment]
PTAL
Now using platform.
Also removed `skipOnTravis()`, it was not used and the check didn't seem 100% right anyway. When we need it back, we can think of a better approach.

File: test/_init_tests.js [hunk 2]
```diff
@@ -43,36 +43,37 @@ class TestConfig {
 
   constructor(runner) {
     this.runner = runner;
-    this.skippedUserAgents = [];
+    /**
+     * List of predicate functions that are called before running each test
+     * suite to check whether the suite should be skipped or not.
+     * If any of the functions return 'true', the suite will be skipped.
+     * @type {!Array<function>}
```
Comment (erwinmombay) [inline test/_init_tests.js]: please add `()` should be type `{!Array<function()>}`

Comment (jridgewell) [inline test/_init_tests.js]: `function():boolean`

**Comment by lannka** [comment]
LGTM

**Comment by jridgewell** [comment]
:shipit:

File: test/_init_tests.js [hunk 3]
```diff
@@ -55,25 +55,26 @@ class TestConfig {
      * @type {!Array<function(!TestSuite)>}
      */
     this.configTasks = [];
+    this.platform_ = platformFor(window);
```
Comment (jridgewell) [inline test/_init_tests.js]: ❤️😍

Comment (aghassemi) [inline test/_init_tests.js]: lol  designed a fashionable T-Shirt for you:

lztrksjlqhq

Comment (jridgewell) [inline test/_init_tests.js]: `s/$/s/`. 😜

**Pull request merged**